docs/source/en/optimization/coreml.mdx

```diff
@@ -0,0 +1,164 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# How to run Stable Diffusion natively on Mac and iPhone
+
+There are two ways to run Stable Diffusion on Apple hardware:
+
+- Using the `diffusers` Python codebase and leveraging PyTorch's `mps` device. This is suitable for Macs running Apple Silicon chips (M1 or M2), and is explained in [the `mps` guide](mps).
+
+- Converting Stable Diffusion models to Core ML, the model format supported by Apple frameworks, and using the converted models in native apps. This works on all types of Apple hardware, including iPhone, iPad, M1/M2 Macs and even Intel Macs with enough GPU memory.
+
+## Stable Diffusion Core ML Checkpoints
+
+Core ML is the native ML model format for Apple devices. Because Stable Diffusion was trained using PyTorch, the model weights are available in PyTorch format. Therefore, we need to convert them to Core ML before we can use them inside native apps.
+
+Thankfully, Apple engineers developed [a conversion tool](https://github.com/apple/ml-stable-diffusion) based on `diffusers`. If you have a Mac, you can use the conversion scripts it provides to convert any Stable Diffusion model to Core ML.
+
+Before you convert a model, though, do take a moment to explore the Hugging Face Hub – chances are the model you're interested in is already available in Core ML format. For example, these official models are available in Core ML in the Apple organization:
+
+- Stable Diffusion v1.4: [converted](https://hf.co/apple/coreml-stable-diffusion-v1-4) [original](https://hf.co/CompVis/stable-diffusion-v1-4)
+- Stable Diffusion v1.5: [converted](https://hf.co/apple/coreml-stable-diffusion-v1-5) [original](https://hf.co/runwayml/stable-diffusion-v1-5)
+- Stable Diffusion v2 base: [converted](https://hf.co/apple/coreml-stable-diffusion-2-base) [original](https://huggingface.co/stabilityai/stable-diffusion-2-base)
+- Stable Diffusion v2.1 base: [converted](https://hf.co/apple/coreml-stable-diffusion-2-1-base) [original](https://huggingface.co/stabilityai/stable-diffusion-2-1-base)
```

Comment:
Maybe organize this info in a table so its a bit easier to read. Or it might be even simpler to just include a link to Apple's org and have the user check it out there instead of listing them here.

Reply:
+1